File: Leetcode/Typescript/1267. Count Servers that Communicate/solution.ts
// https://leetcode.com/problems/count-servers-that-communicate/description

function countServers(grid: number[][]): number {
    const serverRow = Array.from({length: grid.length}, _ => 0)
    const serverCol = Array.from({length: grid[0].length}, _ => 0)

    for (let y = 0; y < grid.length; y++){
        for (let x = 0; x < grid[0].length; x++){
            if (grid[y][x] === 1){
                serverRow[y] += 1
                serverCol[x] += 1
            }
        }
    }

    let connected = 0

    for (let y = 0; y < grid.length; y++){
        for (let x = 0; x < grid[0].length; x++){
            if (grid[y][x] === 1 && (serverRow[y] > 1 || serverCol[x] > 1)){
                connected += 1
            }
        }
    }

    return connected
};

File: Leetcode/Python/1267. Count Servers that Communicate/solution.py
# https://leetcode.com/problems/count-servers-that-communicate/description

class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        server_col, server_row = [0] * len(grid[0]), [0] * len(grid)

        for y in range(len(grid)):
            for x in range(len(grid[0])):
                if grid[y][x]:
                    server_row[y] += 1
                    server_col[x] += 1

        connected = 0

        for y in range(len(grid)):
            for x in range(len(grid[0])):
                if grid[y][x] and (server_row[y] > 1 or server_col[x] > 1):
                    connected += 1

        return connected
